frame/staking/src/voter_bags.rs

```diff
@@ -347,9 +352,26 @@ impl<T: Config> VoterList<T> {
 			},
 			"all `bag_upper` in storage must be members of the new thresholds",
 		);
+		debug_assert!(Self::sanity_check());
 
 		num_affected
 	}
+
+	// Sanity check the voter list.
+	fn sanity_check() -> bool {
+		// Check:
+		// 1) Iterate all voters in list and make sure there are no duplicates.
+		let mut seen_in_list = BTreeSet::new();
+		assert!(Self::iter().map(|node| node.voter.id).all(|voter| seen_in_list.insert(voter)));
+		// 2) Iterate all voters and ensure their count is in sync with `VoterCount`.
+		assert_eq!(Self::iter().collect::<sp_std::vec::Vec<_>>().len() as u32, crate::VoterCount::<T>::get());
+
+		// NOTE: we don't check `CounterForNominators + CounterForValidators == VoterCount`
```

Comment:
Better to also check, but move sanity_check to the outer `lib.rs`. You probably need to make it a free function for that, not an associated one.

Reply:
I think we already have a few debug_asserts for `CounterForNominators + CounterForValidators == VoterCount`. You can replace them all with this.